src/routes/acquirerRoutes.ts

```diff
@@ -48,18 +47,41 @@ const extractFormData = (stepData: Step, body: RequestBody ) => {
     } 
   }
 
+  if (stepData.id === 'date') {
+    return {
+      day: body.day || null,
+      month: body.month || null,
+      year: body.year || null
+    };
+}
   // Other input types can go here
   return
 }
 
+function isValidDate(day: string, month: string, year: string): boolean {
+  // If all fields are empty, consider it valid since the date is optional
+  if (!day && !month && !year) return true;
+
+  // Ensure day, month, year are numbers
+  if (isNaN(Number(day)) || isNaN(Number(month)) || isNaN(Number(year))) return false;
+
+  const d = Number(day);
+  const m = Number(month) - 1; // Month is 0-indexed (0 for January, 11 for December)
+  const y = Number(year);
+  // Use the Date object to create a date
+  const date = new Date(y, m, d);
+
+  // Validate if the created date matches the input values
+  return date && date.getMonth() === m && date.getDate() === d && date.getFullYear() === y;
+}
+
```

Comment:
We might want to do a bit of validation of the day, month and year values. 
I think the `new Date` constructor allows you to use pretty much any values for the year, month and day. E.g. 
```
> new Date(2023, 55, 88)
2027-10-26T23:00:00.000Z
```
So `new Date(2023, 55, 88)` give you a date which is 55 months and 88 days after 1st Jan 2023. Which probably isn't what we want..!

There must be some libraries that would do date validation for us. I'll have a quick look tomorrow.